include/deal.II/base/vectorization.h

```diff
@@ -39,6 +39,22 @@
 // 'check_01_cpu_features.cmake', ensures that these feature are not only
 // present in the compilation unit but also working properly.
 
+#if DEAL_II_COMPILER_VECTORIZATION_LEVEL >= 1 && not defined(__SSE2__)
+#error "Mismatch in vectorization capabilities"
+#error "SSE2 was detected during configuration but it is not available in the compilation unit"
```

Comment:
That's a good suggestion. I've changed it to this (and put everything on a single line because it shows up much nicer in the compiler error message).